packages/node-sdk/sm/primitives/evm/rpc/paima-l2.ts

```diff
@@ -251,10 +277,26 @@ export default function* processPaimaL2SyncProtocolResponse(
       }
     }
   } else { // !isBatched
+    // Direct (non-batched) nonce as hash of [blockHeight, userAddress, game input]
+    let directNonce: string | undefined = undefined;
+    try {
+      const gameInputStr = hexToString(outerLayerData.data);
+      const userAddress =
+        CryptoManager.Evm().verifyAddress(outerLayerData.userAddress)
+          ? Value.Decode(
+            TypeboxHelpers.Evm.Address,
+            outerLayerData.userAddress,
+          )
+          : outerLayerData.userAddress;
+      const toHash = String(paima_block_height) + userAddress + gameInputStr;
+      directNonce = keccak256(stringToHex(toHash));
+    } catch {
+      /* ignore decode failure; skip dedup for this input */
+    }
```

Comment:
In the block processing we cannot have unknown states, let it throw an exception if it must, but it should never happen (remove the try catch, and if there is a specific condition that fails, let's capture and manage it)